File: felicia/examples/learn/service/common/cc/service_create_flag.cc
#include "felicia/examples/learn/service/common/cc/service_create_flag.h"

namespace felicia {

ServiceCreateFlag::ServiceCreateFlag() {
  {
    BoolFlag::Builder builder(MakeValueStore(&is_server_));
    auto flag = builder.SetLongName("--server")
                    .SetHelp(
                        "create server node, if set, default: false(create "
                        "client node)")
                    .Build();
    is_server_flag_ = std::make_unique<BoolFlag>(flag);
  }
  {
    StringFlag::Builder builder(MakeValueStore(&service_));
    auto flag = builder.SetShortName("-s")
                    .SetLongName("--service")
                    .SetHelp("name for service, it's required to be set")
                    .Build();
    service_flag_ = std::make_unique<StringFlag>(flag);
  }
}

ServiceCreateFlag::~ServiceCreateFlag() = default;

bool ServiceCreateFlag::Parse(FlagParser& parser) {
  return PARSE_OPTIONAL_FLAG(parser, name_flag_, is_server_flag_,
                             service_flag_);
}

bool ServiceCreateFlag::Validate() const {
  return CheckIfFlagWasSet(service_flag_);
}

}  // namespace felicia

File: felicia/examples/learn/service/common/python/service_create_flag.py
from felicia.core.protobuf.channel_pb2 import ChannelDef
from felicia.examples.learn.common.python.node_create_flag import NodeCreateFlag
import felicia_py.command_line_interface as cli


class ServiceCreateFlag(NodeCreateFlag):
    def __init__(self):
        super().__init__()
        self.is_server_flag = cli.BoolFlagBuilder().set_long_name("--server").set_help(
            "create server node, if set, default: false(create client node)").build()
        self.service_flag = cli.StringFlagBuilder().set_short_name("-s").set_long_name("--service").set_long_name(
            "--service").set_help("name for service, it's required to be set").build()

    def parse(self, flag_parser):
        return self.parse_optional_flags(flag_parser)

    def validate(self):
        return self.check_if_flag_was_set(self.service_flag)
